src/activities/assignments/AssignmentEntity.js

```diff
@@ -9,12 +10,52 @@ export class AssignmentEntity extends Entity {
 	name() {
 		return this._entity && this._entity.properties && this._entity.properties.name;
 	}
+
 	canEditName() {
-		return this._entity && this._entity.hasActionByName('update-name');
+		return this._entity && this._entity.hasActionByName(Actions.assignments.updateName);
 	}
+
 	getSaveNameAction() {
-		return this._entity && this._entity.getActionByName('update-name');
+		return this._entity && this._entity.getActionByName(Actions.assignments.updateName);
+	}
+
+	_getInstructionsEntity() {
+		return this._entity
+			&& this._entity.hasSubEntityByRel(Rels.Assignments.instructions)
+			&& this._entity.getSubEntityByRel(Rels.Assignments.instructions);
 	}
 
+	instructionsPlaintext() {
+		const instructionsEntity = this._getInstructionsEntity();
+		return instructionsEntity
+			&& instructionsEntity.properties
+			&& instructionsEntity.properties.text;
+	}
+
+	instructionsHtml() {
+		const instructionsEntity = this._getInstructionsEntity();
+		return instructionsEntity
+			&& instructionsEntity.properties
+			&& instructionsEntity.properties.html;
+	}
+
+	instructionsEditorHtml() {
+		const updateInstructionsAction = this.getSaveInstructionsAction();
+		return updateInstructionsAction
+			&& updateInstructionsAction.hasFieldByName('instructions')
+			&& updateInstructionsAction.getFieldByName('instructions').value;
+	}
+
+	canEditInstructions() {
+		const instructionsEntity = this._getInstructionsEntity();
+		return instructionsEntity
+			&& instructionsEntity.hasActionByName(Actions.assignments.updateInstructions);
+	}
+
+	getSaveInstructionsAction() {
+		const instructionsEntity = this._getInstructionsEntity();
+		return instructionsEntity
+			&& instructionsEntity.getActionByName(Actions.assignments.updateInstructions);
```

Comment:
Should we be checking `hasActionByName` before doing `getActionByName`?

Reply:
Not necessarily, and I'd err on the side of not doing it - for the most part it's the same, but the result of `getSaveInstructionsAction` I think is more-normal this way for actions that don't exist.

If we add the `hasActionByName` check and the action doesn't exist,

```js
const saveAction = getSaveInstructionsAction() // == false
```

whereas the current approach results in

```js
const saveAction = getSaveInstructionsAction() // == undefined
```

which seems more predictable to me. An `undefined` action is pretty clearly "there is no such action", whereas a `false` action is less-clear to me (albeit I'll admit, still fairly obvious). That said, I'm open to be convinced otherwise, as I realize this is a minor thing (but one that has come up before, hence having this position/explanation ready to go 😄)

Reply:
Sure, makes sense.